queries/java.scm

```diff
@@ -0,0 +1,311 @@
+
+;;!! if () {}
+;;!  ^^^^^^^^
+(
+    (if_statement
+        "if" @branch.start
+        condition: (_) @condition
+        consequence: (block) @branch.end
+    ) @condition.domain
+    (#not-parent-type? @condition.domain "if_statement")
+    (#child-range! @condition 0 -1 true true)
+)
+
+;;!! else if () {}
+;;!  ^^^^^^^^^^^^^
+(if_statement
+    "else" @branch.start @condition.domain.start
+    alternative: (if_statement
+        condition: (_) @condition
+        consequence: (block) @branch.end @condition.domain.end
+        (#child-range! @condition 0 -1 true true)
+    )
+)
+
+;;!! else {}
+;;!  ^^^^^^^
+(if_statement
+    "else" @branch.start
+    alternative: (block) @branch.end
+)
+
+;;!! for (int i = 0; i < 5; ++i) {}
+;;!                  ^^^^^
+;;!  ------------------------------
+(for_statement
+    condition: (_) @condition
+) @_.domain
+
+;;!! while (value) {}
+;;!         ^^^^^
+;;!  ----------------
+(while_statement
+    condition: (_) @condition
+    (#child-range! @condition 0 -1 true true)
+) @_.domain
+
+(do_statement
+    condition: (_) @condition
+    (#child-range! @condition 0 -1 true true)
+) @_.domain
+
+;;!! switch (value) {}
+;;!          ^^^^^
+;;!  -----------------
+(switch_expression
+    condition: (_) @switchStatementSubject
+    (#child-range! @switchStatementSubject 0 -1 true true)
+) @_.domain
+
+;;!! true ? 1 : 2
+(ternary_expression
+    condition: (_) @condition
+    consequence: (_) @branch
+) @condition.domain
+(ternary_expression
+    alternative: (_) @branch
+)
+
+;;!! true ? 1 : 2
+;;!  ^^^^^^^^^^^^
+(ternary_expression) @branch.iteration
+
+;;!! void myFunk(int value) {}
+;;!                  ^^^^^
+;;!  -------------------------
+(formal_parameters
+    (formal_parameter
+        (identifier) @name
+    ) @_.domain
+) @_.iteration
+
+;;!! int value = 0;
+;;!      ^^^^^
+;;!  --------------
+(assignment_expression
+    left: (_) @name
+) @_.domain
+
+;;!! Map<String, String>
+;;!     ^^^^^^^  ^^^^^^
+(type_arguments
+    (type_identifier) @type
+)
+
+;;!! List<String> list = value;
+;;!  ^^^^^^^^^^^^
+;;!  --------------------------
+(local_variable_declaration
+    type: (_) @type
+) @_.domain
+
+;;!! name = new ArrayList<String>();
+;;!             ^^^^^^^^^^^^^^^^^
+;;!         -----------------------
+(object_creation_expression
+    type: (_) @type
+) @_.domain
+
+;;!! name = new int[5];
+;;!             ^^^
+;;!         ----------
+(array_creation_expression
+    type: (_) @type
+) @_.domain
+
+;;!! void myFunk(int value) {}
+;;!              ^^^
+;;!              ---------
+(formal_parameter
+    type: (_) @type
+) @_.domain
+
+;;!! int size() {}
+;;!  ^^^
+;;!  -------------
+(method_declaration
+    type: (_) @type
+) @_.domain
+
+;;!! new test()
+;;!  ^^^^^^^^
+;;!  ----------
+(object_creation_expression
+    (argument_list) @functionCallee.end.startOf
+) @functionCallee.start.startOf @_.domain
+
+;;!! new test().bar();
+;;!  ^^^^^^^^^^^^^^
+;;!  ----------------
```

Comment:
does this actually exclude the `;`? If so, we should prob try to include that

Reply:
fixed